File: microservers/immobilierservice/src/main/java/io/multi/immobilierservice/dto/ContactCreateRequest.java
package io.multi.immobilierservice.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

/**
 * <b>Auth obligatoire</b> sur l'endpoint POST /immo/proprietes/{uuid}/contact.
 *
 * <p>Décision (Phase 10a) : JWT requis, pas d'accès anonyme. Sans cela,
 * un bot peut bombarder 24/7 n'importe quel vendeur, transformant la
 * plateforme en passerelle de spam. Le rate-limit IP + captcha qui
 * permettraient l'anonyme sont hors scope pour l'instant.
 *
 * <p>Conséquence : <b>aucun champ téléphone/email/nom</b> dans ce DTO.
 * Le service les déduit du JWT via {@code UserClient.getUserById()} —
 * pas de spoofing possible.
 */
@Data
public class ContactCreateRequest {

    @NotBlank
    @Size(max = 2000)
    private String message;

    @Pattern(regexp = "INFO|VISITE|OFFRE",
            message = "typeDemande doit être INFO, VISITE ou OFFRE")
    private String typeDemande = "INFO";
}

File: mobile_app/lib/features/immo/models/contact_create_request.dart
/// DTO de requête pour `POST /immo/proprietes/{uuid}/contact`.
///
/// Conforme à `ContactCreateRequest.java` côté backend :
/// - `message` : non-blank, ≤2000 chars
/// - `typeDemande` : INFO | VISITE | OFFRE (défaut INFO)
///
/// PAS de champs nom/téléphone/email — déduits du JWT côté backend pour
/// éviter le spoofing (cf. ContactCreateRequest doc Java).
class ContactCreateRequest {
  final String message;
  final String typeDemande;

  const ContactCreateRequest({
    required this.message,
    this.typeDemande = 'INFO',
  });

  Map<String, dynamic> toJson() => {
        'message': message,
        'typeDemande': typeDemande,
      };
}
